frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala

```diff
@@ -357,13 +404,16 @@ class BspMetalsClientSpec(
           """.stripMargin
   )
 
-  private def assertSemanticdbFileFor(sourceFileName: String, state: TestState): Unit = {
+  private def assertSemanticdbFileFor(
+      sourceFileName: String,
+      state: TestState,
+      isPresent: Boolean = true
+  ): Unit = {
     val projectA = state.build.getProjectFor("A").get
     val classesDir = state.client.getUniqueClassesDirFor(projectA, forceGeneration = true)
     val sourcePath = if (sourceFileName.startsWith("/")) sourceFileName else s"/$sourceFileName"
-    assertIsFile(
-      classesDir.resolve(s"META-INF/semanticdb/A/src/$sourcePath.semanticdb")
-    )
+    val file = classesDir.resolve(s"META-INF/semanticdb/A/src/$sourcePath.semanticdb")
+    if (isPresent) assertIsFile(file) else assertNotFile(file)
```

Comment:
Nit: I propose we introduce a new method `assertNotSemanticdbFileFor` that checks the non-existence of a semanticdb file, just as we do with the combo `assertIsFile` and `aasertNotFile`.